library.c

```diff
@@ -3496,6 +3496,47 @@ redis_key_prefix(RedisSock *redis_sock, char **key, size_t *key_len) {
     return 1;
 }
 
+/* This is very similar to PHP >= 7.4 zend_string_concat2 only we are taking
+ * two zend_string arguments rather than two char*, size_t pairs */
+static zend_string *redis_zstr_concat(zend_string *prefix, zend_string *suffix) {
+    zend_string *res;
+    size_t len;
+
+    ZEND_ASSERT(prefix != NULL && suffix != NULL);
+
+    len = ZSTR_LEN(prefix) + ZSTR_LEN(suffix);
+    res = zend_string_alloc(len, 0);
+
+    memcpy(ZSTR_VAL(res), ZSTR_VAL(prefix), ZSTR_LEN(prefix));
+    memcpy(ZSTR_VAL(res) + ZSTR_LEN(prefix), ZSTR_VAL(suffix), ZSTR_LEN(suffix));
+    ZSTR_VAL(res)[len] = '\0';
```

Comment:
Is there a room allocated for this byte?

Reply:
Yeah, if you dig into `zend_string_alloc` it uses `_ZSTR_STRUCT_SIZE(len)` to calculate then length, and if you dig into that, you'll see a +1

https://github.com/php/php-src/blob/296f76453c075d9899044e4a4f74a608dcd6cc6b/Zend/zend_string.h#L148-L150

https://github.com/php/php-src/blob/296f76453c075d9899044e4a4f74a608dcd6cc6b/Zend/zend_string.h#L90

Reply:
Thanks 👍